firewall/privacy_mapper.go

```diff
@@ -812,6 +841,654 @@ func handleUpdatePolicyResponse(db firewalldb.PrivacyMapDB,
+	r *lnrpc.WalletBalanceResponse) (proto.Message, error) {
+
+	return func(_ context.Context, r *lnrpc.WalletBalanceResponse) (
+		proto.Message, error) {
+
+		totalBalance, err := maybeHideAmount(
+			flags, randIntn, r.TotalBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		confirmedBalance, err := maybeHideAmount(
+			flags, randIntn, r.ConfirmedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		unconfirmedBalance, err := maybeHideAmount(
+			flags, randIntn, r.UnconfirmedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		lockedBalance, err := maybeHideAmount(
+			flags, randIntn, r.LockedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		reservedBalanceAnchorChan, err := maybeHideAmount(
+			flags, randIntn, r.ReservedBalanceAnchorChan,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		accountBalance := make(
+			map[string]*lnrpc.WalletAccountBalance,
+			len(r.AccountBalance),
+		)
+		for k, v := range r.AccountBalance {
+			confirmed, err := maybeHideAmount(
+				flags, randIntn, v.ConfirmedBalance,
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			unconfirmed, err := maybeHideAmount(
+				flags, randIntn, v.UnconfirmedBalance,
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			accountBalance[k] = &lnrpc.WalletAccountBalance{
+				ConfirmedBalance:   confirmed,
+				UnconfirmedBalance: unconfirmed,
+			}
+		}
+
+		return &lnrpc.WalletBalanceResponse{
+			TotalBalance:              totalBalance,
+			ConfirmedBalance:          confirmedBalance,
+			UnconfirmedBalance:        unconfirmedBalance,
+			LockedBalance:             lockedBalance,
+			ReservedBalanceAnchorChan: reservedBalanceAnchorChan,
+			AccountBalance:            accountBalance,
+		}, nil
+	}
+}
+
+func handlePendingChannelsResponse(db firewalldb.PrivacyMapDB,
+	flags session.PrivacyFlags,
+	randIntn func(int) (int, error)) func(ctx context.Context,
+	r *lnrpc.PendingChannelsResponse) (proto.Message, error) {
+
+	return func(_ context.Context, r *lnrpc.PendingChannelsResponse) (
+		proto.Message, error) {
+
+		pendingOpen := make(
+			[]*lnrpc.PendingChannelsResponse_PendingOpenChannel,
+			len(r.PendingOpenChannels),
+		)
+
+		err := db.Update(func(tx firewalldb.PrivacyMapTx) error {
+			for i, c := range r.PendingOpenChannels {
+				var err error
+
+				remotePub := c.Channel.RemoteNodePub
+				if !flags.Contains(session.ClearPubkeys) {
+					remotePub, err = firewalldb.HideString(
+						tx, remotePub,
+					)
+					if err != nil {
+						return err
+					}
+				}
+
+				capacity, err := maybeHideAmount(
+					flags, randIntn, c.Channel.Capacity,
+				)
+				if err != nil {
+					return err
+				}
+
+				localBalance, err := maybeHideAmount(
+					flags, randIntn, c.Channel.LocalBalance,
+				)
+				if err != nil {
+					return err
+				}
+
+				remoteBalance, err := maybeHideAmount(
+					flags, randIntn, c.Channel.RemoteBalance,
+				)
+				if err != nil {
+					return err
+				}
```

Comment:
Should we cap the obfuscated value of this to the sum of of the capacity? We could also make the sum of these become equal to the capacity similar to the listchannels response.